src/main/cpp/src/utils/vid_mapper_sql.cc

```diff
@@ -0,0 +1,183 @@
+/*
+ * The MIT License (MIT)
+ * Copyright (c) 2016 Intel Corporation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#ifdef LIBDBI
+#include "vid_mapper_sql.h"
+#include <ctype.h>
+#include <cstdlib>
+#include <cstdio>
+
+#define VERIFY_OR_THROW(X) if(!(X)) throw SQLBasedVidMapperException(#X);
+
+SQLBasedVidMapper::SQLBasedVidMapper(const SQLVidMapperRequest& request) : VidMapper() {
+  dbi_initialize_r(NULL, &m_instance);
+  m_conn = dbi_conn_new_r("pgsql", m_instance);
+  VERIFY_OR_THROW(m_conn != NULL);
+
+  dbi_conn_set_option(m_conn, "host", request.host_name.c_str());
+  dbi_conn_set_option(m_conn, "username", request.user_name.c_str());
+  dbi_conn_set_option(m_conn, "password", request.pass_word.c_str());
+  dbi_conn_set_option(m_conn, "dbname", request.db_name.c_str());
+  dbi_conn_set_option(m_conn, "encoding", "UTF-8");
+
+  if (dbi_conn_connect(m_conn) < 0) {
+    VERIFY_OR_THROW(1 < 0);
```

Comment:
throw exception with a descriptive error message "Could not connect to DB <host>:<port> with <username>"